moto/ec2/models/spot_requests.py

```diff
@@ -202,6 +203,16 @@ def __init__(
 
         self.launch_specs = []
 
+        self.tags = {}
+        if tag_specifications is not None:
+            tags = convert_tag_spec(tag_specifications)
+            for resource_type in tags:
+                if resource_type != "spot-fleet-request":
+                    raise ValueError(
```

Comment:
It's good to add this check, but it also needs to be covered by a test.

Here is an example of how to do that: https://github.com/getmoto/moto/blob/c9d62e94025fb504881510278150d97d9c4dc002/tests/test_ec2/test_spot_instances.py#L34